sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -248,11 +256,13 @@ static int sof_card_late_probe(struct snd_soc_card *card)
 static const struct snd_kcontrol_new sof_controls[] = {
 	SOC_DAPM_PIN_SWITCH("Headphone Jack"),
 	SOC_DAPM_PIN_SWITCH("Headset Mic"),
+	SOC_DAPM_PIN_SWITCH("Spk"),
 };
 
 static const struct snd_soc_dapm_widget sof_widgets[] = {
 	SND_SOC_DAPM_HP("Headphone Jack", NULL),
 	SND_SOC_DAPM_MIC("Headset Mic", NULL),
+	SND_SOC_DAPM_SPK("Spk", NULL),
```

Comment:
can this be added only when there is an amp? Alternatively the card name needs to be changed so that we can build UCM files that don't include an output that's not handled?

Reply:
Since its a static array, i couldn't get how we can add it only in case of SPK available.
For dapm route, we have API to update later using snd_soc_dapm_add_routes(). But similar API is not available for widget. 

In earlier KBL m/c , different card names are used and 2 separate static arrays are defined. I didnt want to use that method. let me know if that is absolutely needed.